domain/credentials/service.go

```diff
@@ -96,6 +100,78 @@ func (s Service) GetCredentials(customer domain.Customer) ([]domain.Organization
 	return credentials, nil
 }
 
+func (s Service) GetCredentialIssuers(credentials []string) (domain.CredentialIssuers, error) {
+	result := domain.CredentialIssuers{}
+	for _, credential := range credentials {
+
+		trustedDIDs, err := s.fetchCredentialIssuers(credential, s.client().ListTrusted)
+		if err != nil {
+			return result, err
+		}
+		untrustedDIDs, err := s.fetchCredentialIssuers(credential, s.client().ListUntrusted)
+		if err != nil {
+			return result, err
+		}
+		issuers := make([]domain.CredentialIssuer, len(trustedDIDs)+len(untrustedDIDs))
+		for i, id := range trustedDIDs {
+			issuer, err := s.GetIssuer(id)
+			if err != nil {
+				return result, err
+			}
+			issuers[i] = domain.CredentialIssuer{Trusted: true, ServiceProvider: *issuer}
+		}
+		for i, id := range untrustedDIDs {
+			issuer, err := s.GetIssuer(id)
+			if err != nil {
+				return result, err
+			}
+			issuers[len(trustedDIDs)+i] = domain.CredentialIssuer{Trusted: false, ServiceProvider: *issuer}
+		}
+		result.Set(credential, issuers)
+	}
+	return result, nil
+}
+
+func (s Service) GetIssuer(id ssi.URI) (*domain.ServiceProvider, error) {
+	sp := &domain.ServiceProvider{Id: id.String()}
+	if id.Scheme != "did" {
+		return sp, nil
+	}
+	contactInformation, err := s.SPService.DIDManClient.GetContactInformation(sp.Id)
+	if err != nil {
+		return nil, unwrapAPIError(err)
+	}
+	if contactInformation != nil {
+		sp.Email = contactInformation.Email
+		sp.Name = contactInformation.Name
+		sp.Phone = contactInformation.Phone
+		sp.Website = contactInformation.Website
+	}
+	return sp, nil
+}
+
+func (s Service) fetchCredentialIssuers(credential string, clientFn func(ctx context.Context, credentialType string) (*http.Response, error)) ([]ssi.URI, error) {
+	var issuerDIDs []ssi.URI
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Sounds like a refactor for a different PR

Reply:
Because the wrong client API type is used, you're using the "plain" OpenAPI generated one, rather than the friendlier-to-use `vcr.api.HTTPClient` we wrote ourselves (which wraps the one which is used here, `vcr.api.ClientInterface`)

Reply:
I see the nice one is missing funcs, which will be added later on